packages/editor-ui/src/components/TagsContainer.vue

```diff
@@ -1,133 +1,124 @@
-<script lang="ts">
-import { defineComponent, type ComponentInstance } from 'vue';
-
+<script lang="ts" setup>
+import { ref, computed, onMounted, onBeforeUnmount, nextTick } from 'vue';
+import type { ComponentInstance } from 'vue';
 import type { ITag } from '@/Interface';
 import IntersectionObserver from './IntersectionObserver.vue';
 import IntersectionObserved from './IntersectionObserved.vue';
-import { mapStores } from 'pinia';
-import { useTagsStore } from '@/stores/tags.store';
 import { createEventBus } from 'n8n-design-system/utils';
 import { debounce } from 'lodash-es';
 
-// random upper limit if none is set to minimize performance impact of observers
-const DEFAULT_MAX_TAGS_LIMIT = 20;
-
-interface TagEl extends ITag {
-	hidden?: boolean;
-	title?: string;
-	isCount?: boolean;
+// Define the interface for the props
+interface TagsContainerProps {
+	tagIds: string[];
+	tagsById: { [id: string]: ITag };
+	limit?: number;
+	clickable?: boolean;
+	responsive?: boolean;
+	hoverable?: boolean;
 }
 
-export default defineComponent({
-	name: 'TagsContainer',
-	components: { IntersectionObserver, IntersectionObserved },
-	props: {
-		tagIds: {
-			type: Array as () => string[],
-			required: true,
-		},
-		limit: {
-			type: Number,
-			default: DEFAULT_MAX_TAGS_LIMIT,
-		},
-		clickable: Boolean,
-		responsive: Boolean,
-		hoverable: Boolean,
-	},
-	emits: {
-		click: null,
-	},
-	data() {
-		return {
-			maxWidth: 320,
-			intersectionEventBus: createEventBus(),
-			visibility: {} as { [id: string]: boolean },
-			debouncedSetMaxWidth: () => {},
+// Define props using the interface and withDefaults
```

Comment:
Just a personal opinion but I don't think this comment and the one above provide much value, as it's pretty obvious what's happening